### PR TITLE
Feature: Run test using already running local_binary

### PR DESCRIPTION
**Pull request by yashLadha**
* Run tests using already running local_binary through identifier
* Added env to controll api polling time interval.

PS: In case user has passed indentifier that is not running, them tests
won't be able to complete and IDLE_TIMEOUT occurs. This is an addon and
user should use it as per their understanding.

### CODE REVIEW

**Comment by yashLadha** [comment]
@AndreyBelym Can you please review this.

File: src/backends/automate.js [hunk 1]
```diff
@@ -6,7 +6,14 @@ import requestApiBase from '../utils/request-api';
...
 
-const API_POLLING_INTERVAL = 80000;
+// This is done to prevent idle timeout happening
+// in automate sessions, as network can become bottleneck.
+// Try reducing the number in case network is unable to
+// send commands in the given amount of time
+const envAPIPoll = process.env['API_POLL'];
+const API_POLLING_INTERVAL = !isNaN(envAPIPoll) ?
+                            parseInt(envAPIPoll, 10) :
+                            80000;
```
Comment (AndreyBelym) [inline src/backends/automate.js]: I suggest checking the value if it is lesser than [BROWSERSTACK_IDLE_TIMEOUT](https://www.browserstack.com/automate/timeouts) and emit a warning with [process.emitWarning](https://nodejs.org/api/process.html#process_process_emitwarning_warning_type_code_ctor) in this case.

File: src/index.js [hunk 2]
```diff
@@ -188,6 +196,12 @@ export default {
             capabilities.localIdentifier = connector.connectorInstance.localIdentifierFlag;
             capabilities.local           = true;
         }
+        else {
+            // Can utilise benefit of already running binary on
+            // the system
+            capabilities.local = true;
```
Comment (AndreyBelym) [inline src/index.js]: This flag is duplicated, could we avoid duplicating it somehow?

File: src/index.js [hunk 3]
```diff
@@ -171,8 +171,16 @@ export default {
...
-        var connector    = await this._getConnector();
+        var capabilities = {
+            ...this._generateBasicCapabilities(browserName),
+            ...this._getAdditionalCapabilities()
+        };
+        const localIdentifier = process.env['BROWSERSTACK_LOCAL_ID'];
+
+        let connector = null;
+
+        if (!localIdentifier)
```
Comment (AndreyBelym) [inline src/index.js]: If you don't create a connector, you should avoid disposing `null` here:
https://github.com/DevExpress/testcafe-browser-provider-browserstack/blob/master/src/index.js#L217

Comment (yashLadha) [reply src/index.js]: It will be resolved to null only. Is there any harm in this approach?

Comment (AndreyBelym) [reply src/index.js]: Nevermind, I forgot that the check for `null` is implemented already: https://github.com/DevExpress/testcafe-browser-provider-browserstack/blob/master/src/index.js#L47

File: src/backends/automate.js [hunk 4]
```diff
@@ -6,7 +6,14 @@ import requestApiBase from '../utils/request-api';
...
 import * as ERROR_MESSAGES from '../templates/error-messages';
 
-const API_POLLING_INTERVAL = 80000;
+// This is done to prevent idle timeout happening
+// in automate sessions, as network can become bottleneck.
+// Try reducing the number in case network is unable to
+// send commands in the given amount of time
+const envAPIPoll = process.env['API_POLL'];
+const API_POLLING_INTERVAL = !isNaN(envAPIPoll) ?
+                            parseInt(envAPIPoll, 10) :
```
[the remark this answers is not in the document]
Comment (yashLadha) [reply src/backends/automate.js]: Done

File: src/backends/automate.js (hunk 1 above)
Comment (yashLadha) [reply src/backends/automate.js]: Done

File: src/index.js (hunk 2 above)
Comment (yashLadha) [reply src/index.js]: Moved

**Comment by yashLadha** [comment]
@AndreyBelym can this be merged?

**Comment by yashLadha** [comment]
@VasilyStrelyaev Can you please review?

**Comment by yashLadha** [comment]
@AndreyBelym Can this be merged. It is been hanging from a log time.

**Comment by yashLadha** [comment]
@AndreyBelym

**Comment by yashLadha** [comment]
@VasilyStrelyaev

**Comment by AndreyBelym** [comment]
Thank you for your contribution. I will post a message here when a new version with the changes will be released.

**Comment by yashLadha** [comment]
Thanks @AndreyBelym

**Comment by AndreyBelym** [comment]
Thank you for your patience, `testcafe-browser-provider-browserstack@1.12.0` is available now.

**Pull request merged**